test/e2e/network_peering_test.go

```diff
@@ -229,63 +195,53 @@ var _ = Describe("NetworkPeering", Label("networkpeering"), func() {
 
 })
 
-func DeleteAllNetworkPeering(data *model.TestDataProvider) {
+func DeleteAllNetworkPeering(testData *model.TestDataProvider) {
 	errList := make([]error, 0)
-	project, err := kube.GetProjectResource(data)
-	Expect(err).ToNot(HaveOccurred())
-	errors := networkpeer.DeletePeerVPC(project.Status.NetworkPeers)
+	Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{Name: testData.Project.Name, Namespace: testData.Project.Namespace}, testData.Project)).ToNot(HaveOccurred())
+	errors := networkpeer.DeletePeerVPC(testData.Project.Status.NetworkPeers)
 	errList = append(errList, errors...)
 	Expect(errList).To(BeEmpty())
 }
 
 func networkPeerFlow(userData *model.TestDataProvider, peers []v1.NetworkPeer) {
-	By("Deploy Project with requested configuration", func() {
-		actions.PrepareUsersConfigurations(userData)
-		deploy.NamespacedOperator(userData)
-		actions.DeployProjectAndWait(userData, "1")
-	})
-
 	By("Prepare network peers cloud infrastructure", func() {
 		err := networkpeer.PreparePeerVPC(peers)
 		Expect(err).ToNot(HaveOccurred())
 	})
 
 	By("Create network peers", func() {
-		for _, peer := range peers {
-			userData.Resources.Project.WithNetworkPeer(peer)
-		}
-		actions.PrepareUsersConfigurations(userData)
-		actions.DeployProject(userData)
+		Expect(userData.K8SClient.Get(userData.Context, types.NamespacedName{Name: userData.Project.Name,
+			Namespace: userData.Project.Namespace}, userData.Project)).Should(Succeed())
+		userData.Project.Spec.NetworkPeers = append(userData.Project.Spec.NetworkPeers, peers...)
+		Expect(userData.K8SClient.Update(userData.Context, userData.Project)).Should(Succeed())
 	})
 
 	By("Establish network peers connection", func() {
 		Eventually(func() bool {
 			return EnsurePeersReadyToConnect(*userData, len(peers))
 		}).WithTimeout(5*time.Minute).WithPolling(20*time.Second).Should(BeTrue(), "Network Peering should be ready to establish connection")
-		project, err := kube.GetProjectResource(userData)
-		Expect(err).ShouldNot(HaveOccurred())
-		err = networkpeer.EstablishPeerConnections(project.Status.NetworkPeers)
-		Expect(err).ShouldNot(HaveOccurred())
-		Eventually(kube.GetProjectNetworkPeerStatus(userData), "1m", "20s").Should(Equal("True"), "NetworkPeerStatus should be True")
+		Expect(userData.K8SClient.Get(userData.Context, types.NamespacedName{Name: userData.Project.Name, Namespace: userData.Project.Namespace}, userData.Project)).Should(Succeed())
+		Expect(networkpeer.EstablishPeerConnections(userData.Project.Status.NetworkPeers)).Should(Succeed())
+		Eventually(func() string {
```

Comment:
`Eventually` here should be with Gomega context, as `GetProjectNetworkPeeringStatus` uses `Expect` matcher